Replace eawsy/aws-lambda-go with aws-lambda-go

paulolai opened the pull request:
Replace github.com/eawsy/aws-lambda-go/service/lambda/runtime with github.com/aws/aws-lambda-go/lambda This now determines if it's running as a Lambda when the program is launched.

Also add in a makefile, update the documentation and an example tf file.

I've tested that this works for cloudwatch when ran locally and when ran as an AWS Lambda.

lox commented:
Thanks @paulolai! My main concern here is change of `handler.handle`. I wonder if we should keep that the same for consistency with previous versions?

paulolai commented:
Oh, I didn't see those scripts there! I agree, there should only be one
source of truth.

On Wed, Oct 17, 2018 at 3:27 PM Lachlan Donald <notifications@github.com>
wrote:

> I'd be fine to have a Makefile be the source of truth, but probably worth
> copying over some of the nuances from
> https://github.com/buildkite/buildkite-agent-metrics/blob/master/.buildkite/steps/build-binary.sh
> such as how the binary is carefully built to be static
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/buildkite/buildkite-agent-metrics/pull/58#issuecomment-430485381>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AnYPoS0D7qASqO8aS0OCs90hsD7P-_mCks5ulrGXgaJpZM4XjBxT>
> .
>

-- 
*P.S. We've launched a new blog to share the latest ideas and case studies 
from our team. Check it out here: product.canva.com 
<http://product.canva.com/>. ***
** <https://canva.com>Empowering the world 
to design
Also, we're hiring. Apply here! 
<https://about.canva.com/careers/>
 <https://twitter.com/canva> 
<https://facebook.com/canva> <https://au.linkedin.com/company/canva> 
<https://instagram.com/canva>

paulolai commented:
@lox I've removed the Makefile, have switched back to handler.handle (note that you will still need to change it from python2.7 to go1.x ), updated the build scripts and bumped up the version number.

I've removed build_lambda.sh as I removed eawsy/aws-lambda-go, instead I modified build_binaries.sh to manually create the package (renaming then zipping it).

build_binaries.sh now no longer takes arguments, instead it always builds the darwin and linux arm64 binaries and creates the lambda package. It only requires BUILDKITE_BUILD_NUMBER to have been set. You will need to change your tooling to account for that.

You normally use cloudformation, I don't know enough about that so I've included the TF file I've written for my own use as a reference, you should update your cloudformation file (update the runtime to go1.x from python2.7).

I don't have buildkite-agent on our system to upload the artifacts, you'll need to test that part of the scripts yourself.

paulolai commented:
This is replaced by https://github.com/buildkite/buildkite-agent-metrics/pull/59
